langwatch/src/components/AddTriggerDrawer.tsx

```diff
@@ -0,0 +1,260 @@
+import {
+  Button,
+  Checkbox,
+  Drawer,
+  DrawerBody,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerHeader,
+  FormControl,
+  FormErrorMessage,
+  HStack,
+  Input,
+  Popover,
+  PopoverBody,
+  PopoverCloseButton,
+  PopoverContent,
+  PopoverTrigger,
+  Radio,
+  RadioGroup,
+  Stack,
+  Text,
+  VStack,
+  useDisclosure,
+  useToast,
+} from "@chakra-ui/react";
+import { TriggerAction } from "@prisma/client";
+import { useDrawer } from "~/components/CurrentDrawer";
+
+import { HorizontalFormControl } from "./HorizontalFormControl";
+
+import { useFilterParams } from "~/hooks/useFilterParams";
+
+import { useForm } from "react-hook-form";
+import { useOrganizationTeamProject } from "~/hooks/useOrganizationTeamProject";
+import { api } from "~/utils/api";
+
+export function TriggerDrawer() {
+  const { project, organization } = useOrganizationTeamProject();
+  const { onOpen, onClose, isOpen } = useDisclosure();
+
+  const toast = useToast();
+  const createTrigger = api.trigger.create.useMutation();
+
+  const organizationWithMembers =
+    api.organization.getOrganizationWithMembersAndTheirTeams.useQuery(
+      {
+        organizationId: organization?.id ?? "",
+      },
+      { enabled: !!organization }
+    );
+
+  const { closeDrawer } = useDrawer();
+
+  const { filterParams } = useFilterParams();
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    getValues,
+    formState: { errors },
+    reset,
+  } = useForm({
+    defaultValues: {
+      name: "",
+      action: TriggerAction.SEND_EMAIL,
+      email: "",
+      members: [],
+    },
+  });
+
+  const currentAction = watch("action");
+
+  const onSubmit = (data: any) => {
```

Comment:
could you add the type here instead of `any`? Check out how other `onSubmit`s do, I think more recently on the add dataset record drawer I changed, you can define the form type up above and use the same here